### ISSUE TITLE
Make Upgrade Modules consumable

### ISSUE THREAD

**Issue by ThePiachu** [StrangeLoopGames/EcoSuggestions#1288]
Recently on the Feedback channel I talked about the economy on White Tiger and the conversation reached a conclusion that it would be nice if the various Upgrade Modules were consumables.

In a similar fashion to #1077, this would help to create a constant demand for even lower-tier products. The idea would be that you would be able to slot an Upgrade in anywhere, but then it would be consumed by that table and provide the bonus for some amount of crafting time (say, 24 hours). You could add more Modules into the slot to consume them when the previous ones run out, etc.

This would make it so players can't juggle one good module between all their workstations, as well as creating a constant demand for the Modules, which means they would be supplied more readily on the market (instead of the market drying up of BU4s when everyone moves into T2 era, leaving some newbies without any upgrades).

It would also hopefully circulate some steady stream of money back from say, cooks to manufacturers since they would need to keep re-buying the upgrade modules.

It would also help some newer players to contribute to the economy with this sink - modules are rather easy to transport, so they could buy whichever ones they need, upgrade them one step in their home and then bring the upgraded ones back in a backpack.

Someone suggested they could be repaired as tools instead, perhaps requiring a lower-tier Upgrade as a repair component, but that would only be interesting if the repair mechanic was revamped like in #1257. Manual repairs are so tedious...

**Comment by mirasrael** [comment]
We discussed idea of consumables modules like Steel Saw for T4 stone processing (it is actually why we have steel in recipe for "virtual saw"). So idea of consumables modules in general is good and applicable in many scenarios (but it is more like a fuel in this case).

For upgrade modules though if we will make them consumable then we have to significantly reduce it's price. Also instead of solving the issue it may make it more problematic with increasing module demand and accidental module expiration for big craft. So it may introduce more issues than solve.

As alternative for issue with module juggling we can make modules non detachable (in this case when you pickup table it will preserve installed modules). Also as partial solution with module deficit we can change how modules crafted. Instead of crafting BU2 from BU1 it may be crafted independent (so no more need to organize chain for module production), but instead it may be used for upgrading table with BU1. In this case any upgrade producer may produce as many modules as he want no matter how much previous tier modules exists. They just can't be used on tables without previous tier upgrade. It won't change principal efficiency chain, but will help with market fulfilling.

As additional option we can allow to produce some upgrades of same level for multiple professions. But we should be very careful here to avoid situation when one-two professions may be used to skip progress for fast modules leveling. To avoid this we can use another mechanic we previously discussed - Technologies. So instead of locking modules to professions we can lock modules to Technology which locked to profession level. Once technology discovered it unlocks upgrades craft. So you don't need to ask and wait people for upgrades, but instead you may craft it by your own if you know technology (it may be much more expensive than for professional thought).

**Comment by johnkslg** [comment]
I would like to add a lot more maintainence/repair to things in the game, and having consumed modules could be a great way to do that.

**Comment by mirasrael** [comment]
I like consumables modules idea in general, but it shouldn't be applied to Upgrades. It is fine to have consumables modules which *doesn't affect efficiency*, but required for production, like Steel Saw for T4 Stone processing.
But having consumable upgrades will make production much more complex and stressful. There is few points which comes on top of my mind:

1. Main difference with fuel or required module you should care about queuing it up by your own and it will create stress for some people (like me). Because if I'm out of fuel then it will just stop production and I can come and fix it. If upgrade expired then it will continue production, but with low efficiency. And I may end up with negative profit (i.e. I set price to 10 and craft with module was 9, but after module expired self-cost become 11). So I should always calculate how modules I need for all my crafts and should find, buy and queue them in upwards.
2. You can't calculate it as 1/10 or 1/20 of current module price unless you have module with life-time like 30 days. But in this case it doesn't make much sense to even make them consumable. With shorter life-time they just become another kind of fuel and so should be included in self-cost of goods. With infinite life-time you can ignore it in self-cost, because on long period it is neglectable. But with part of self-cost it creates many strange situations:
- It increases cost for all goods, because now they should include module cost;
- For some goods it may be more profitable to produce them with lower efficiency module or without module at all, because added cost from module is more than from saved ingredients. And it may be even more confusing if you have to produce some profitable goods and non-profitable goods on same table, but both of them will use same module.
3. This was mainly suggested for fixing issue with modules availability on market, but it can make it even worse. Now if no modules on market you may at least continue production on upgraded tables, but if you out of consumable modules then it may stop all production.

@johnkslg @ThePiachu Don't you think this change will make too negative impact on gameplay?

**Comment by Engimage** [comment]
I mentioned this in Discord several times.
Modules can be replaced with "Tools" of different tier/quality. These tools would be used as a fuel for the station, and depending on the tier of the tool used the station gets "module" benefits.

Basic Engineer/Mechanic/industialist can be the one actually producing all grades of tools. They can utilize different kinds of materials provided by other professions. This is opposed to spreading the module recipes between random professions actually dictating their usefullness because of that. Anf engineers actually lose their demand pretty fast and they don't actually have any materials they should provide on a regular basis. Don't count the roads as they are not related to engineering at all.

**Comment by mirasrael** [comment]
I added suggestion for technologies: https://github.com/StrangeLoopGames/EcoSuggestions/issues/1300 it describes alternative way how issue with modules may be fixed as part of proposal.

**Comment by elegorod** [comment]
Interesting idea, but directly making consumable modules like fuel will lead to issues:
1. Module ends in the middle of the craft. When I started crafting 400 Barrels on another player's table with module 2, I want to make sure that all barrels will be crafted with module 2. Crafting with no module will be money loss. Mb module durability can be consumed when craft is started and returned back when craft is cancelled.
2. On servers with low rates (now I play on x0,3 exp rate plus need 4 times more exp to level up specialties) it takes several days to level up Basic engineering 6, a week for Mechanics 6, two weeks for Industry 4. If player who has Mechanics 6 will stop playing the game, all server will be without modules 4 and 5 at all for whole week! Even worse, when another player will take Mechanics star, he'll need thousands or Iron bars. But smelter can't cheaply make them because modules 4-5 are no longer available. Looks like economic collapse. This can be fixed with server-wide technologies proposed in #1300 . When first player on server reaches Mechanics 6, other players need just Mechanics 1 to craft the module.
3. Players will fear that module will be broken soon and won't set cheaper prices when they get new module. This can lead to overpricing, especially in smelting, mechanics, industry. Would be good that module stays with me for at least few days, so I'm safe to decrease sale prices without risking to go bankrupt.

**Comment by ThePiachu** [comment]
My baseline thinking for a module was for it to last 24 active hours of production. You usually don't produce things in that big quantities in one go. At level 5 upgrade, that's ~4000 steel plates, ~750 asphalt roads, 20k hewn logs. Anything that consumes fuel would probably run out of fuel way before you run out of a Module. So with this kind of efficiency you wouldn't even need to alter the price of the modules since they would be largely negligible. Of course, with the modules lasting a shorter amount of time it might be an issue.

What I personally found on a few runs was that whatever isn't in demand won't be supplied. Day 14 on a server you might have no BU1s, 2s or 3s being made because nobody is buying BU4s, until they run out in a few days and then you have to bug people to get the production back up and running. And yeah, someone quitting would have a similar effect.

When it comes to prices changing because some modules are unavailable, that to some extent is offset with at least some raw material production maturing during the game. Day 1 logs are expensive because you only have so many trees, while day 7 you already have entire forests planted and ready to harvest for cheap.

But I do get where people are coming from - efficiency is a very important part of production in ECO, plus loss-aversion is another compounding factor on top of it.

Consumable modules are one way of making sure there is a constant need for a lot of crafting professions. Some components that need replacing would be another, but those might disproportionally favour things like smelting, mechanics, etc. (how many components would require stuff from a mason or a logger?).

**Comment by Engimage** [comment]
I wanna give some answers to @mirasrael criticism.

1. Regarding module expiration in the middle of the craft. I would say the craft should stop just like when the fuel depletes in other tables. But there should be a 0% efficiency module that everyone can craft themselves at the toolbench. So there will not be a situation that the craft became unpredictably more expensive. Maybe not by themselves but consider it as having "expendable spare parts" on the table. You always buy those with the machinery knowing it has a limited lifetime.
2. Module prices might be severely decreased letting everyone use them before they can actually save enough resources to make multiple full cost modules as they are now. Getting 5 "fuel" modules at 1/10 cost will let a person queue like a day of production on 5 tables as opposed to the price of crafting 5 full modules. And I do not think this will have a major impact on the item price if their price is reduced. It is quite similar to existing hand tools. Those are pretty expendable (at some point). Just calculate how many units of your wares you can craft with 1 fuel module and you will see it is less than 1% of the price.
3. With the modules spread between professions you can actually lose even one (any) of them to prevent availability of the modules. If you give all modules to a dedicated profession like an engineer those will become pretty important and valuable specialists, and you can probably have at least 2 of them per server for redundancy. This is much more reliable compared to existing scheme. Also in its current state Basic Engineer and Mechanic are the hardest to level and they absolutely lose their demand after the modules are installed for people.

**Comment by mirasrael** [comment]
Yeah, if all these 3 conditions will be implemented it may work, but there another problems appears.
1. 0% efficiency module looks like a hack. And it means all tables which worked before out of box and can use upgrade modules will now require some "fuel" even for basic crafts. As alternative you may select upgrade level you wanna to use for the table and it will automatically consume upgrade modules if their available, but stop if no more modules. It may auto select when you put upgrade module in module cell and you can clear it up explicitly if it expired and you wanna to continue craft without module.
2. If we are speaking about 24h usage time for module then yes, price may be neglectable. But then 1/10 of price may be too cheap. Need a good balance for this.
3. I like an idea of having single person like engineer to craft all modules. It is much easier to ensure at least one such profession on server and it makes this profession very desired. It also adds mass product to the profession. But it breaks gating system we have for enforcing parallel professions development. It may be fixed though with Technologies I suggested in another ticket. Another problem is that having single person responsible for modules can make them too powerful and rich.

To summarize. I think this system may work if:
- it will stop when module consumed;
- it should have long enough time (like 24h+) and cheap enough price. We should also deny to remove module as soon as it start consuming.
- we should have alternative gating system (like Technologies).

**Comment by Engimage** [comment]
@mirasrael 
* You can put more than one spare part into your table to guarantee continuous production.
* Once the module is placed the first one is consumed (same as with fuel) so no need to invent something to prevent the removal
* 0% efficiency module is fine. You can't go mining without a pickaxe right? Same here. It might have a simple recipe but still...
* powerful and rich - talk about chefs. It is actually cool to make ALL of the professions actually important. And the more important you make them the stronger economy becomes. Yes its a problem that some profession still lack on this side but it is exactly something to address, not other way around. 

@johnkslg 
P.S.
Make higher level recipes require certain level of housing (or at least the room comfort, not only tier) and voila - builders and housing item producers are important. Make recipes require a certain level of food points to perform labor - and voila, farmers and chefs are required.
